xgi/classes/function.py

```diff
@@ -903,3 +905,71 @@ def subfaces(edges, order=None):
 
         faces += faces_to_add
     return faces
+
+
+def complement(H):
+    """Returns the complement of hypergraph H.
+
+    The complement of an hypergraph will have the same nodes (same indexes) and
+    will contain every possible hyperedge linking these nodes, provided it is
+    not already present in the hypergraph and it is not larger than its largest
+    hyperedge.
+    Hyperedges of size one are taken into account.
```

Comment:
```suggestion
    The complement Hc of a hypergraph H has the same nodes (same indexes) as H and 
    contains every possible hyperedge linking these nodes, except those present in H. 
    The maximum hyperedge size in Hc is the same as in H.
    
    Singleton edges are taken into account.
```
```suggestion
    The complement of an hypergraph will have the same nodes (same indexes) and
    will contain every possible hyperedge linking these nodes, provided it is
    not already present in the hypergraph and it is not larger than its largest
    hyperedge.
    Hyperedges of size one are taken into account.
```

Reply:
I used your description, it is clearer !